lib/constructs/interface.js

```diff
@@ -111,9 +111,21 @@ Interface.prototype.generateConstructor = function () {
     this.str += `
   iface.setup(this${passArgs});
 }\n`;
+  } else if (this.name === 'HTMLElement') {
+    this.str += `function ${this.name}() {
+  const { document, customElements } = global.window;
```

Comment:
hmm, I've run into this myself.  I ended up having to add so much crap to simulate the window object I created its own module and would pull it in as a dependency